backend/charm/src/faults.py

```diff
@@ -0,0 +1,18 @@
+class UnitReadinessFault:
+    """A ReadinessFault describes the reason for a unit not to be ready."""
+
+    def __init__(self, reason: str, parent=None):
```

Comment:
I experienced a Python noob issue here that I worked around by ignoring type hinting the `parent` here: I tried adding a type hint `parent=UnitReadinessFault | None` here but VS Code with mypy running didn't appreciate that. I also tried stringifying it, which made VS Code happy but caused a runtime issue with `"str" | None` being taken at charm runtime to be the type.